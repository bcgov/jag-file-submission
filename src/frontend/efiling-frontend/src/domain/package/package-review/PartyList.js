import React from "react";
import PropTypes from "prop-types";
import { formatFullName } from "../../../modules/helpers/StringUtil";
import "./PartyList.scss";

const hash = require("object-hash");

export default function PartyList({ parties }) {
  return (
    <div className="ct-party-list">
      <div className="header">
        <span className="d-none d-lg-inline col-lg-6">Name</span>
        <span className="d-none d-lg-inline col-lg-3">Party Role</span>
        <span className="d-none d-lg-inline col-lg-3">Party Type</span>
      </div>
      <ul>
        {parties &&
          parties.map((party) => (
            <li key={hash(party)}>
              <span className="label col-sm-4 d-lg-none">Name:</span>
              <span className="col-sm-8 col-lg-6">{formatFullName(party)}</span>
              <span className="label col-sm-4 d-lg-none">Party Role:</span>
              <span className="col-sm-8 col-lg-3">{party.roleDescription}</span>
              <span className="label col-sm-4 d-lg-none">Party Type:</span>
              <span className="col-sm-8 col-lg-3">
                {party.partyDescription}
              </span>
            </li>
          ))}
      </ul>
    </div>
  );
}

PartyList.propTypes = {
  parties: PropTypes.arrayOf(
    PropTypes.shape({
      partyType: PropTypes.string,
      partyDescription: PropTypes.string,
      roleType: PropTypes.string,
      roleDescription: PropTypes.string,
      firstName: PropTypes.string,
      middleName: PropTypes.string,
      lastName: PropTypes.string,
    })
  ),
};

PartyList.defaultProps = {
  parties: [],
};
