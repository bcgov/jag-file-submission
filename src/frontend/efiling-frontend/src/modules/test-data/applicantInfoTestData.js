const applicantInfo = {
  bceid: "bobross42",
  firstName: "Bob",
  lastName: "Ross",
  email: "bob.ross@example.com",
};

export function getApplicantInfo() {
  return applicantInfo;
}
