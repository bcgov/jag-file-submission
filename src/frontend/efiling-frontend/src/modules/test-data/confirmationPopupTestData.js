/* eslint-disable react/jsx-one-expression-per-line */
import React from "react";

const onButtonClick = () => {};

const mainButton = {
  label: "Cancel",
  styling: "bcgov-normal-white btn",
  onClick: onButtonClick,
};

const confirmButton = {
  label: "Yes, cancel",
  styling: "bcgov-normal-blue btn bcgov-consistent-width",
  onClick: onButtonClick,
};

const cancelButton = {
  label: "No, dont cancel",
  styling: "bcgov-normal-white btn bcgov-consistent-width",
  onClick: onButtonClick,
};

const body = () => (
  <>
    <p>Your files will not be submitted.</p>
    <p>
      You will be returned to:
      <br />
      <b>Original</b> website
    </p>
  </>
);

export function getTestData() {
  const modal = {
    show: false,
    title: "title",
    body,
  };

  return {
    modal,
    mainButton,
    confirmButton,
    cancelButton,
  };
}
