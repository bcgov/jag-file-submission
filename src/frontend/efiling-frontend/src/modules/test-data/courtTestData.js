const court = {
  locationDescription: "Court location",
  fileNumber: "Court file number",
  levelDescription: "Level",
  classDescription: "Class",
  level: "P",
  courtClass: "F",
  location: "Kelowna Law Courts",
};

export function getCourtData() {
  return court;
}
