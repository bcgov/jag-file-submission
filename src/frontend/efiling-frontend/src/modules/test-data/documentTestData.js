const documents = [
  {
    documentProperties: {
      name: "file name 1",
      type: "file type",
    },
    description: "file description 1",
    statutoryFeeAmount: 40,
    isAmendment: null,
    isSupremeCourtScheduling: null,
  },
  {
    documentProperties: {
      name: "file name 2",
      type: "file type",
    },
    description: "file description 2",
    statutoryFeeAmount: 0,
    isAmendment: false,
    isSupremeCourtScheduling: true,
  },
];

export function getDocumentsData() {
  return documents;
}
