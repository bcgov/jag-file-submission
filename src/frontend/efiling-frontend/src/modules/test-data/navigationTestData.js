const navigationUrls = {
  cancel: "cancelurl.com",
  success: "successurl.com",
  error: "",
};

export function getNavigationData() {
  return navigationUrls;
}
