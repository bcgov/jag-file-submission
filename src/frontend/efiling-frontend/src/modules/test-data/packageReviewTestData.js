export const documents = [
  {
    identifier: "1",
    documentProperties: {
      type: "AFF",
      name: "test-document.pdf",
    },
    status: {
      description: "Submitted",
    },
    filingDate: "2020-05-05T00:00:00.000Z",
  },
];

export const parties = [
  {
    partyType: "IND",
    roleType: "APP",
    firstName: "Bob",
    middleName: "Q",
    lastName: "Ross",
    roleDescription: "Applicant",
    partyDescription: "Individual",
  },
];

export const payments = [
  {
    feeExempt: false,
    paymentCategory: 1,
    processedAmount: 7,
    submittedAmount: 7,
    serviceIdentifier: 0,
    transactionDate: null,
  },
];
