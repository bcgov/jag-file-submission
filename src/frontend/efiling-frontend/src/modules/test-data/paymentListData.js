export const payments = [
  {
    feeExempt: false,
    paymentCategory: 1,
    processedAmount: 7,
    submittedAmount: 7,
    serviceIdentifier: 0,
    transactionDate: null,
  },
  {
    feeExempt: true,
    paymentCategory: 3,
    processedAmount: 7,
    submittedAmount: null,
    serviceIdentifier: 0,
    transactionDate: null,
  },
  {
    feeExempt: false,
    paymentCategory: 3,
    processedAmount: 7,
    submittedAmount: 7,
    serviceIdentifier: 0,
    transactionDate: null,
  },
  {
    feeExempt: false,
    paymentCategory: 3,
    processedAmount: null,
    submittedAmount: 7,
    serviceIdentifier: 0,
    transactionDate: null,
  },
];
