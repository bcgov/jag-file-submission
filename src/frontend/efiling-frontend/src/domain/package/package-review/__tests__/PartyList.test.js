import React from "react";
import { render, waitFor } from "@testing-library/react";
import PartyList from "../PartyList";
import * as packageReviewTestData from "../../../../modules/test-data/packageReviewTestData";

describe("PartyList Component", () => {
  beforeEach(() => {});

  test("Matches snapshot", () => {
    const { parties } = packageReviewTestData;
    const { asFragment } = render(<PartyList parties={parties} />);
    waitFor(() => {});

    expect(asFragment()).toMatchSnapshot();
  });
});
