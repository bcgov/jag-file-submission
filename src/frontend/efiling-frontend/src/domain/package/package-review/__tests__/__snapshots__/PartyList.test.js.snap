// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PartyList Component Matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ct-party-list"
  >
    <div
      class="header"
    >
      <span
        class="d-none d-lg-inline col-lg-6"
      >
        Name
      </span>
      <span
        class="d-none d-lg-inline col-lg-3"
      >
        Party Role
      </span>
      <span
        class="d-none d-lg-inline col-lg-3"
      >
        Party Type
      </span>
    </div>
    <ul>
      <li>
        <span
          class="col-sm-12 col-lg-6"
        >
          <svg
            color="#FCBA19"
            fill="currentColor"
            height="30"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(252, 186, 25);"
            viewBox="0 0 24 24"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 5.9c1.16 0 2.1.94 2.1 2.1s-.94 2.1-2.1 2.1S9.9 9.16 9.9 8s.94-2.1 2.1-2.1m0 9c2.97 0 6.1 1.46 6.1 2.1v1.1H5.9V17c0-.64 3.13-2.1 6.1-2.1M12 4C9.79 4 8 5.79 8 8s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 9c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4z"
            />
          </svg>
          Ross, Bob Q
        </span>
        <span
          class="label col-sm-4 d-lg-none"
        >
          Party Role:
        </span>
        <span
          class="col-sm-8 col-lg-3"
        >
          Applicant
        </span>
        <span
          class="label col-sm-4 d-lg-none"
        >
          Party Type:
        </span>
        <span
          class="col-sm-8 col-lg-3"
        >
          Individual
        </span>
      </li>
      <li>
        <span
          class="col-sm-12 col-lg-6"
        >
          <svg
            enable-background="new 0 0 48 48"
            fill="currentColor"
            height="30"
            stroke="currentColor"
            stroke-width="0"
            version="1"
            viewBox="0 0 48 48"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M42,42H6V10c0-2.2,1.8-4,4-4h28c2.2,0,4,1.8,4,4V42z"
              fill="#90CAF9"
            />
            <rect
              fill="#64B5F6"
              height="2"
              width="36"
              x="6"
              y="42"
            />
            <g
              fill="#1565C0"
            >
              <rect
                height="5"
                width="6"
                x="31"
                y="27"
              />
              <rect
                height="5"
                width="6"
                x="21"
                y="27"
              />
              <rect
                height="5"
                width="6"
                x="11"
                y="27"
              />
              <rect
                height="5"
                width="6"
                x="31"
                y="35"
              />
              <rect
                height="5"
                width="6"
                x="11"
                y="35"
              />
              <rect
                height="5"
                width="6"
                x="31"
                y="19"
              />
              <rect
                height="5"
                width="6"
                x="21"
                y="19"
              />
              <rect
                height="5"
                width="6"
                x="11"
                y="19"
              />
              <rect
                height="5"
                width="6"
                x="31"
                y="11"
              />
              <rect
                height="5"
                width="6"
                x="21"
                y="11"
              />
              <rect
                height="5"
                width="6"
                x="11"
                y="11"
              />
              <rect
                height="9"
                width="6"
                x="21"
                y="35"
              />
            </g>
          </svg>
          The Q Continuum
        </span>
        <span
          class="label col-sm-4 d-lg-none"
        >
          Party Role:
        </span>
        <span
          class="col-sm-8 col-lg-3"
        >
          Applicant
        </span>
        <span
          class="label col-sm-4 d-lg-none"
        >
          Party Type:
        </span>
        <span
          class="col-sm-8 col-lg-3"
        >
          Organization
        </span>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
